docs/cross-compilation.md

```diff
@@ -0,0 +1,63 @@
+# How to analyze a project that contains cross-compilation 
+
+It may happen that due to special gcc build-time configuration (see https://gcc.gnu.org/onlinedocs/libstdc++/manual/configure.html) clang cannot do the analysis.
+This is especially the case when gcc is used for **cross-compilation**.
+
+**In case of cross-compilation always use the same target for clang analysis as you use in the gcc build.** 
+
+A file (`commonhandler.cpp`) for example compiles like this to powerpc target:
+``` 
+powerpc64-wrs-linux-g++ -c -o /home/<username>/mylib_ppc64_linux/commonhandler.cpp
+```
+when we try to analyze this file with Clang Static Analyzer with the same command line options like
+``` 
+ CodeChecker quickcheck -b "powerpc64-wrs-linux-g++ -c -o /home/<username>/mylib_ppc64_linux/commonhandler.cpp" --analyzers clangsa --verbose debug
+```
+we get the following error:
+```
+ .../platform_ifc/RTLIB.h:42: 
+ /proj/.../inc/myprod_te_lib.h:60:10: fatal error: 'lib.h' file not found
+ #include <lib.h>                 /* Original lib.h           */        
+ 1 error generated.
+```
+Why? Because `powerpc64-wrs-linux-g++` is an product specific cross compiler and has some compiled in include paths, and defines, while clang, that was delivered with CodeChecker is configured to compile to x86_64 linux target.
```

Comment:
It is possible that the user builds his package in that case we do not deliver clang in the CodeCecker package.